README.md

```diff
@@ -79,6 +76,30 @@ class CNN(nn.Module):
     return x
 ```
 
+```py
+class AutoEncoder(Module):
+  encoder_widths: Iterable
+  decoder_widths: Iterable
+  input_shape: Tuple = None
+
+  def setup(self):
+    self.encoder = MLP(self.encoder_widths)
+    self.decoder = MLP(self.decoder_widths + (jnp.prod(self.input_shape, ))
+
+  def __call__(self, x):
+    return self.decode(self.encode(x))
+
+  def encode(self, x):
+    assert x.shape[-len(self.input_shape):] == self.input_shape
```

Comment:
shouldn't we check
```
    assert x.shape[1:] == self.input_shape
```

if there are more than 1 dimensions in front of the expected `self.input_shape`, they will be flattened in the next line, and not maintained in the return value from `decode()`

(e.g. if `input_shape=(3, 4)` and `x.shape=(5,10,3,4)` then the return value from `decode()` will have `shape=(50,3,4)`)

Reply:
good catch.